src/app/app.component.ts

```diff
@@ -339,6 +339,45 @@ export class AppComponent implements OnInit {
               this.router.navigate(["vault"]);
             }
             break;
+          case "systemSuspended":
+            for (const userId in this.stateService.accounts.getValue()) {
+              if (userId == null) {
+                continue;
+              }
+              const options = await this.getVaultTimeoutOptions(userId);
+              if (options[0] === -3) {
+                options[1] === "logOut"
+                  ? this.messagingService.send("logout", { expired: false, userId: userId })
+                  : this.messagingService.send("lockVault", { userId: userId });
+              }
+            }
```

Comment:
This logic is repeated for the systemCommands, with the only difference being the value of options[0] checked. This could be put into a separate method with a param to equal options[0]. 

Going further instead of sending a message to itself, the logic inside of the `logout` and `lockVault` commands could also be put into methods. These could then be called from the `cases` and from the new method proposed above. 
This would obviously only be an option if no other part of the code relies on receiving `logout` and `lockVault` messages.